corva/app/stream.py

```diff
@@ -1,100 +1,96 @@
-            pre_process_kwargs=pre_process_kwargs,
-            process_kwargs=process_kwargs,
-            post_process_kwargs=post_process_kwargs,
-            on_fail_before_post_process_kwargs=on_fail_before_post_process_kwargs
+            state=RedisState(
+                redis=RedisAdapter(
+                    default_name=GetStateKey.from_event(event=event, app_key=self.app_key),
+                    cache_url=self.cache_url,
+                    logger=self.logger
+                ),
+                logger=self.logger
+            )
         )
 
-    def pre_process(self, event: BaseEvent, state: RedisState, **kwargs) -> ProcessResult:
-        event = super().pre_process(event=event, state=state, **kwargs).event
-
+    def pre_process(self, context: StreamContext) -> None:
         last_processed_timestamp = (
-            int(state.load(key='last_processed_timestamp'))
+            int(context.state.load(key='last_processed_timestamp') or self.DEFAULT_LAST_PROCESSED_VALUE)
             if self.filter_by_timestamp
-            else self.DEFAULT_LAST_PROCESSED_TIMESTAMP
+            else self.DEFAULT_LAST_PROCESSED_VALUE
         )
+
         last_processed_depth = (
-            float(state.load(key='last_processed_depth'))
+            float(context.state.load(key='last_processed_depth') or self.DEFAULT_LAST_PROCESSED_VALUE)
             if self.filter_by_depth
-            else self.DEFAULT_LAST_PROCESSED_DEPTH
+            else self.DEFAULT_LAST_PROCESSED_VALUE
         )
 
         event = self._filter_event(
-            event=event,
+            event=context.event,
             last_processed_timestamp=last_processed_timestamp,
             last_processed_depth=last_processed_depth
         )
 
-        return ProcessResult(event=event)
+        context.event = event
 
-    def post_process(self, event: BaseEvent, state: RedisState, **kwargs) -> ProcessResult:
-        event = super(StreamApp, self).post_process(event=event, state=state, **kwargs).event
+    def post_process(self, context: StreamContext) -> None:
+        all_records: List[Record] = list(chain(*[subdata.records for subdata in context.event]))
 
-        all_records: List[Record] = list(chain(*[subdata.records for subdata in event]))
         last_processed_timestamp = max(
             [
                 record.timestamp
                 for record in all_records
                 if record.timestamp is not None
             ],
-            default=self.DEFAULT_LAST_PROCESSED_TIMESTAMP
+            default=self.DEFAULT_LAST_PROCESSED_VALUE
         )
         last_processed_depth = max(
             [
                 record.measured_depth
                 for record in all_records
                 if record.measured_depth is not None
             ],
-            default=self.DEFAULT_LAST_PROCESSED_DEPTH
+            default=self.DEFAULT_LAST_PROCESSED_VALUE
         )
 
         mapping = {'last_processed_timestamp': last_processed_timestamp,
```

Comment:
Not directly related to this PR but no need to store to cache if the value is missing. Depth based data never has timestamp (and vice versa).